Test failure (skipped) for FrontendConnectionErrorScreenTest

<!-- READ THIS FIRST:
- Make sure you run the latest version of the Android app
- Make sure you run the latest version of Home Assistant
- Make sure to check the Companion docs for troubleshooting and configuration: https://companion.home-assistant.io/
- Make sure the bug you found is not already reported, we love to put work in bugfixes instead of closing duplicate bug reports
  DO NOT DELETE ANY TEXT from this template! All requested information is important.
-->

<!-- If you are reporting an issue for Wear OS please list both devices for the below 3 fields -->
**Home Assistant Android app version(s):** ad7da3f47bd81f1f38d2959f5a6e2274b0fb04cb

**Android version(s):** n/a

**Device model(s):** n/a

**Home Assistant version:** n/a

**Last working Home Assistant release (if known):** b000e2b9030d3aac1d3c1d0421dda00b3ddd10c3

**Description of problem, include YAML if issue is related to notifications:**

Unit tests are currently failing because of skipped test(s) for `FrontendConnectionErrorScreenTest`.
 - This _only_ occurs when executed from the command line with `./gradlew :app:testFullDebugUnitTest`, not when executed from Android Studio.
 - This failure can be seen on recent PRs such as #6342, but details are not in the CI output and instead only in the generated file in `/app/build/reports/tests/testFullDebugUnitTest`.
 - It reports one failed test, `Given FrontendConnectionErrorScreen with viewmodel when retry is clicked then connectivity check is retried`, but that is because it immediately stops; the same stack trace is generated for any test in the file if you start removing `@Test` annotations.

```
[CrashFailFastHandler] - ██████████████████████
!!! CRITICAL FAILURE: FAIL-FAST !!!
██████████████████████

An unrecoverable error has occurred, and the FailFast mechanism
has been triggered. The application cannot continue and will now exit.

ACTION REQUIRED: This error must be investigated and resolved.
Review the accompanying stack trace for details.
----------------------------------------------------------------


android.os.strictmode.IncorrectContextUseViolation: The API:LayoutInflater needs a proper configuration. Use UI contexts such as an activity or a context created via createWindowContext(Display, int, Bundle) or  createConfigurationContext(Configuration) with a proper configuration.
	at android.os.strictmode.Violation.$$robo$$android_os_strictmode_Violation$fillInStackTrace(Violation.java:61)
	at android.os.strictmode.Violation.fillInStackTrace(Violation.java)
	at java.base/java.lang.Throwable.<init>(Throwable.java:283)
	at android.os.strictmode.Violation.<init>(Violation.java:25)
	at android.os.strictmode.IncorrectContextUseViolation.<init>(IncorrectContextUseViolation.java:33)
	at android.os.StrictMode.$$robo$$android_os_StrictMode$onIncorrectContextUsed(StrictMode.java:2395)
	at android.os.StrictMode.onIncorrectContextUsed(StrictMode.java)
	at android.os.StrictMode.$$robo$$android_os_StrictMode$assertConfigurationContext(StrictMode.java:2431)
	at android.os.StrictMode.assertConfigurationContext(StrictMode.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$__constructor__(LayoutInflater.java:241)
	at android.view.LayoutInflater.<init>(LayoutInflater.java:240)
	at com.android.internal.policy.PhoneLayoutInflater.<init>(PhoneLayoutInflater.java:44)
	at android.app.SystemServiceRegistry$31.$$robo$$android_app_SystemServiceRegistry_31$createService(SystemServiceRegistry.java:596)
	at android.app.SystemServiceRegistry$31.createService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry$31.createService(SystemServiceRegistry.java:593)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.$$robo$$android_app_SystemServiceRegistry_CachedServiceFetcher$getService(SystemServiceRegistry.java:2343)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry.$$robo$$android_app_SystemServiceRegistry$getSystemService(SystemServiceRegistry.java:1918)
	at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java)
	at android.app.ContextImpl.$$robo$$android_app_ContextImpl$getSystemService(ContextImpl.java:2315)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.robolectric.shadows.ShadowContextImpl$ContextImplReflector$$Reflector5.getSystemService(Unknown Source)
	at org.robolectric.shadows.ShadowContextImpl.getSystemService(ShadowContextImpl.java:83)
	at android.app.ContextImpl.getSystemService(ContextImpl.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$from(LayoutInflater.java:267)
	at android.view.LayoutInflater.from(LayoutInflater.java)
	at android.view.ContextThemeWrapper.$$robo$$android_view_ContextThemeWrapper$getSystemService(ContextThemeWrapper.java:184)
	at android.view.ContextThemeWrapper.getSystemService(ContextThemeWrapper.java)
	at android.app.Activity.$$robo$$android_app_Activity$getSystemService(Activity.java:7896)
	at android.app.Activity.getSystemService(Activity.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$from(LayoutInflater.java:267)
	at android.view.LayoutInflater.from(LayoutInflater.java)
	at com.android.internal.policy.PhoneWindow.$$robo$$com_android_internal_policy_PhoneWindow$__constructor__(PhoneWindow.java:407)
	at com.android.internal.policy.PhoneWindow.<init>(PhoneWindow.java:406)
	at com.android.internal.policy.PhoneWindow.<init>(PhoneWindow.java:420)
	at android.app.Activity.$$robo$$android_app_Activity$attach(Activity.java:9047)
	at android.app.Activity.attach(Activity.java)
	at android.app.Activity.$$robo$$android_app_Activity$attach(Activity.java:9024)
	at android.app.Activity.attach(Activity.java)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.robolectric.shadows.ActivityReflector$$Reflector47.attach(Unknown Source)
	at org.robolectric.shadows.ActivityReflector.callAttach(ActivityReflector.java:225)
	at org.robolectric.shadows.ShadowActivity.callAttach(ShadowActivity.java:216)
	at org.robolectric.android.controller.ActivityController.attach(ActivityController.java:115)
	at org.robolectric.android.controller.ActivityController.attach(ActivityController.java:94)
	at org.robolectric.android.controller.ActivityController.of(ActivityController.java:76)
	at org.robolectric.Robolectric.buildActivity(Robolectric.java:131)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.lambda$startActivitySyncInternal$0(RoboMonitoringInstrumentation.java:128)
	at org.robolectric.shadows.ShadowInstrumentation.runOnMainSyncNoIdle(ShadowInstrumentation.java:1183)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.startActivitySyncInternal(RoboMonitoringInstrumentation.java:125)
	at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:38)
	at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:43)
	at androidx.test.core.app.ActivityScenario.launchInternal(ActivityScenario.java:362)
	at androidx.test.core.app.ActivityScenario.launch(ActivityScenario.java:202)
	at androidx.test.ext.junit.rules.ActivityScenarioRule.lambda$new$0(ActivityScenarioRule.java:77)
	at androidx.test.ext.junit.rules.ActivityScenarioRule$$ExternalSyntheticLambda1.get(D8$$SyntheticClass)
	at androidx.test.ext.junit.rules.ActivityScenarioRule.before(ActivityScenarioRule.java:110)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invokeSuspend(AndroidComposeTestRule.android.kt:334)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invoke(AndroidComposeTestRule.android.kt)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invoke(AndroidComposeTestRule.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invokeSuspend(ComposeUiTest.android.kt:605)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.IdlingStrategy.withStrategy$suspendImpl(IdlingStrategy.android.kt:55)
	at androidx.compose.ui.test.IdlingStrategy.withStrategy(IdlingStrategy.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invokeSuspend(ComposeUiTest.android.kt:602)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invoke(ComposeUiTest.android.kt)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:317)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:24)
	at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:99)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:326)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:10)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:309)
	at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:167)
	at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment.runTest(ComposeUiTest.android.kt:585)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1.evaluate(AndroidComposeTestRule.android.kt:334)
	at dagger.hilt.android.internal.testing.MarkThatRulesRanRule$1.evaluate(MarkThatRulesRanRule.java:108)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:524)
	at org.robolectric.internal.SandboxTestRunner.executeInSandbox(SandboxTestRunner.java:494)
	at org.robolectric.internal.SandboxTestRunner.access$900(SandboxTestRunner.java:67)
	at org.robolectric.internal.SandboxTestRunner$7.evaluate(SandboxTestRunner.java:442)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.robolectric.internal.SandboxTestRunner.access$600(SandboxTestRunner.java:67)
	at org.robolectric.internal.SandboxTestRunner$6.evaluate(SandboxTestRunner.java:333)
	at org.robolectric.internal.SandboxTestRunner$3.evaluate(SandboxTestRunner.java:233)
	at org.robolectric.internal.SandboxTestRunner$5.lambda$evaluate$0(SandboxTestRunner.java:317)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:101)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.IllegalAccessException: Tried to access the API:LayoutInflater which needs to have proper configuration from a non-UI Context:dagger.hilt.android.testing.HiltTestApplication@23c53e29
	at android.os.StrictMode.$$robo$$android_os_StrictMode$assertConfigurationContext(StrictMode.java:2430)
	... 103 more
 - android.os.strictmode.IncorrectContextUseViolation: The API:LayoutInflater needs a proper configuration. Use UI contexts such as an activity or a context created via createWindowContext(Display, int, Bundle) or  createConfigurationContext(Configuration) with a proper configuration.
	at android.os.strictmode.Violation.$$robo$$android_os_strictmode_Violation$fillInStackTrace(Violation.java:61)
	at android.os.strictmode.Violation.fillInStackTrace(Violation.java)
	at java.base/java.lang.Throwable.<init>(Throwable.java:283)
	at android.os.strictmode.Violation.<init>(Violation.java:25)
	at android.os.strictmode.IncorrectContextUseViolation.<init>(IncorrectContextUseViolation.java:33)
	at android.os.StrictMode.$$robo$$android_os_StrictMode$onIncorrectContextUsed(StrictMode.java:2395)
	at android.os.StrictMode.onIncorrectContextUsed(StrictMode.java)
	at android.os.StrictMode.$$robo$$android_os_StrictMode$assertConfigurationContext(StrictMode.java:2431)
	at android.os.StrictMode.assertConfigurationContext(StrictMode.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$__constructor__(LayoutInflater.java:241)
	at android.view.LayoutInflater.<init>(LayoutInflater.java:240)
	at com.android.internal.policy.PhoneLayoutInflater.<init>(PhoneLayoutInflater.java:44)
	at android.app.SystemServiceRegistry$31.$$robo$$android_app_SystemServiceRegistry_31$createService(SystemServiceRegistry.java:596)
	at android.app.SystemServiceRegistry$31.createService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry$31.createService(SystemServiceRegistry.java:593)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.$$robo$$android_app_SystemServiceRegistry_CachedServiceFetcher$getService(SystemServiceRegistry.java:2343)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry.$$robo$$android_app_SystemServiceRegistry$getSystemService(SystemServiceRegistry.java:1918)
	at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java)
	at android.app.ContextImpl.$$robo$$android_app_ContextImpl$getSystemService(ContextImpl.java:2315)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.robolectric.shadows.ShadowContextImpl$ContextImplReflector$$Reflector5.getSystemService(Unknown Source)
	at org.robolectric.shadows.ShadowContextImpl.getSystemService(ShadowContextImpl.java:83)
	at android.app.ContextImpl.getSystemService(ContextImpl.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$from(LayoutInflater.java:267)
	at android.view.LayoutInflater.from(LayoutInflater.java)
	at android.view.ContextThemeWrapper.$$robo$$android_view_ContextThemeWrapper$getSystemService(ContextThemeWrapper.java:184)
	at android.view.ContextThemeWrapper.getSystemService(ContextThemeWrapper.java)
	at android.app.Activity.$$robo$$android_app_Activity$getSystemService(Activity.java:7896)
	at android.app.Activity.getSystemService(Activity.java)
	at android.view.LayoutInflater.$$robo$$android_view_LayoutInflater$from(LayoutInflater.java:267)
	at android.view.LayoutInflater.from(LayoutInflater.java)
	at com.android.internal.policy.PhoneWindow.$$robo$$com_android_internal_policy_PhoneWindow$__constructor__(PhoneWindow.java:407)
	at com.android.internal.policy.PhoneWindow.<init>(PhoneWindow.java:406)
	at com.android.internal.policy.PhoneWindow.<init>(PhoneWindow.java:420)
	at android.app.Activity.$$robo$$android_app_Activity$attach(Activity.java:9047)
	at android.app.Activity.attach(Activity.java)
	at android.app.Activity.$$robo$$android_app_Activity$attach(Activity.java:9024)
	at android.app.Activity.attach(Activity.java)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.robolectric.shadows.ActivityReflector$$Reflector47.attach(Unknown Source)
	at org.robolectric.shadows.ActivityReflector.callAttach(ActivityReflector.java:225)
	at org.robolectric.shadows.ShadowActivity.callAttach(ShadowActivity.java:216)
	at org.robolectric.android.controller.ActivityController.attach(ActivityController.java:115)
	at org.robolectric.android.controller.ActivityController.attach(ActivityController.java:94)
	at org.robolectric.android.controller.ActivityController.of(ActivityController.java:76)
	at org.robolectric.Robolectric.buildActivity(Robolectric.java:131)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.lambda$startActivitySyncInternal$0(RoboMonitoringInstrumentation.java:128)
	at org.robolectric.shadows.ShadowInstrumentation.runOnMainSyncNoIdle(ShadowInstrumentation.java:1183)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.startActivitySyncInternal(RoboMonitoringInstrumentation.java:125)
	at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:38)
	at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:43)
	at androidx.test.core.app.ActivityScenario.launchInternal(ActivityScenario.java:362)
	at androidx.test.core.app.ActivityScenario.launch(ActivityScenario.java:202)
	at androidx.test.ext.junit.rules.ActivityScenarioRule.lambda$new$0(ActivityScenarioRule.java:77)
	at androidx.test.ext.junit.rules.ActivityScenarioRule$$ExternalSyntheticLambda1.get(D8$$SyntheticClass)
	at androidx.test.ext.junit.rules.ActivityScenarioRule.before(ActivityScenarioRule.java:110)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invokeSuspend(AndroidComposeTestRule.android.kt:334)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invoke(AndroidComposeTestRule.android.kt)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1$evaluate$1.invoke(AndroidComposeTestRule.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invokeSuspend(ComposeUiTest.android.kt:605)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.IdlingStrategy.withStrategy$suspendImpl(IdlingStrategy.android.kt:55)
	at androidx.compose.ui.test.IdlingStrategy.withStrategy(IdlingStrategy.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invokeSuspend(ComposeUiTest.android.kt:602)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invoke(ComposeUiTest.android.kt)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invoke(ComposeUiTest.android.kt)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:317)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:24)
	at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:99)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:326)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:10)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:309)
	at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:167)
	at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
	at androidx.compose.ui.test.AndroidComposeUiTestEnvironment.runTest(ComposeUiTest.android.kt:585)
	at androidx.compose.ui.test.junit4.AndroidComposeTestRule$apply$1.evaluate(AndroidComposeTestRule.android.kt:334)
	at dagger.hilt.android.internal.testing.MarkThatRulesRanRule$1.evaluate(MarkThatRulesRanRule.java:108)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:524)
	at org.robolectric.internal.SandboxTestRunner.executeInSandbox(SandboxTestRunner.java:494)
	at org.robolectric.internal.SandboxTestRunner.access$900(SandboxTestRunner.java:67)
	at org.robolectric.internal.SandboxTestRunner$7.evaluate(SandboxTestRunner.java:442)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.robolectric.internal.SandboxTestRunner.access$600(SandboxTestRunner.java:67)
	at org.robolectric.internal.SandboxTestRunner$6.evaluate(SandboxTestRunner.java:333)
	at org.robolectric.internal.SandboxTestRunner$3.evaluate(SandboxTestRunner.java:233)
	at org.robolectric.internal.SandboxTestRunner$5.lambda$evaluate$0(SandboxTestRunner.java:317)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:101)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.IllegalAccessException: Tried to access the API:LayoutInflater which needs to have proper configuration from a non-UI Context:dagger.hilt.android.testing.HiltTestApplication@23c53e29
	at android.os.StrictMode.$$robo$$android_os_StrictMode$assertConfigurationContext(StrictMode.java:2430)
	... 103 more
```

I've tried to fix it but am not making any progress, not enough testing experience yet :(

<!--
- For Wear OS devices we will need the LogCat logs from the device.
- For Android Auto the logs can be retrieved from the connected device.
- Logs from the device can be taken from Settings > Companion App Troubleshooting > Show and Share Logs
-->
**Companion App Logs:** n/a

**Screenshot or video of problem:** n/a

**Additional information:**

--- a/.github/copilot-instructions.md
+++ b/.github/copilot-instructions.md
@@ -305,6 +305,13 @@ Use the `Clock` available from `Hilt`.
 - **Framework**: JUnit Jupiter for unit tests (or 4 when necessary to use Robolectric)
 - **Mocking**: MockK (Use real objects when you can or fakes)
 - **Android APIs**: Robolectric (requires JUnit 4 compatibility)
+- **Robolectric tests**: All Robolectric test classes must use both annotations:
+  ```kotlin
+  @RunWith(RobolectricTestRunner::class)
+  @Config(application = HiltTestApplication::class)
+  class MyTest {
+  ```
+  Without the `@Config` override, Robolectric defaults to the manifest's `HomeAssistantApplication`, which enables StrictMode and FailFast, leaking process-wide state that can crash the test JVM.
 - **Test Location**: Tests should mirror source structure in `src/test/kotlin/` directory
 - **Test Naming**: Use GIVEN-WHEN-THEN structure with descriptive sentences:
   ```kotlin

--- a/app/src/test/kotlin/io/homeassistant/companion/android/launch/link/LinkHandlerTest.kt
+++ b/app/src/test/kotlin/io/homeassistant/companion/android/launch/link/LinkHandlerTest.kt
@@ -1,5 +1,6 @@
 package io.homeassistant.companion.android.launch.link
 
+import dagger.hilt.android.testing.HiltTestApplication
 import androidx.core.net.toUri
 import io.homeassistant.companion.android.common.data.servers.ServerManager
 import io.homeassistant.companion.android.common.util.FailFast
@@ -13,9 +14,11 @@ import org.junit.jupiter.api.assertNotNull
 import org.junit.jupiter.api.fail
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
 
 // We need Robolectric because of the usage of URI from `android.net.URI`
 @RunWith(RobolectricTestRunner::class)
+@Config(application = HiltTestApplication::class)
 class LinkHandlerTest {
 
     private val serverManager: ServerManager = mockk()

--- a/app/src/test/kotlin/io/homeassistant/companion/android/widgets/todo/TodoGlanceAppWidgetTest.kt
+++ b/app/src/test/kotlin/io/homeassistant/companion/android/widgets/todo/TodoGlanceAppWidgetTest.kt
@@ -10,14 +10,17 @@ import androidx.glance.testing.unit.assertHasNoClickAction
 import androidx.glance.testing.unit.hasContentDescriptionEqualTo
 import androidx.glance.testing.unit.hasTestTag
 import androidx.glance.testing.unit.hasTextEqualTo
+import dagger.hilt.android.testing.HiltTestApplication
 import io.homeassistant.companion.android.common.R
 import io.homeassistant.companion.android.database.widget.WidgetBackgroundType
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
 import org.robolectric.RuntimeEnvironment
 
 @RunWith(RobolectricTestRunner::class)
+@Config(application = HiltTestApplication::class)
 class TodoGlanceAppWidgetTest {
     private val context = RuntimeEnvironment.getApplication()
 

